wpengine/atlas-content-modeler
# MTKA-1557: Allow 0 as a value in number fields

## Description

Number fields will now save '0' values instead of discarding them.

Ensures that:
- Required number fields in new entries aren't cleared with a 'this field is required' message when saving them with values of '0'.
- Number fields in existing entries with a non-zero value aren't reset to their original value when editing the field to use '0' and saving the post.

Works for decimal and integer types.

https://wpengine.atlassian.net/browse/MTKA-1557

## Checklist

I have:


## Testing

Extended existing tests to check that 0 values can be saved.

To test manually:

1. Create a required number field.
2. Create a new entry and use `0` for the number field value.
3. Save the post. It should save without errors.
